backend/market/views.py

```diff
@@ -299,5 +308,6 @@ def destroy(self, request, *args, **kwargs):
     def list(self, request, *args, **kwargs):
```

Comment:
Looking at the destroy code on line 303, it seems like you are doing some manual checks to see if the object can be destroyed. Again, I think if we get the permissions right (which I think they are) and the queryset right (which I think you have correct as well), Django's built in destroy should be able to handle all of it for us?